src/mscorlib/src/System/ArraySegment.cs

```diff
@@ -106,12 +110,31 @@ public int Count
             }
         }
 
-        public Enumerator GetEnumerator()
+        public T this[int index]
```

Comment:
+1

This is why ref returns were invented.  Definitely should be used here because there are no compat issues to consider on this new API.